scrunch/cubes.py

```diff
@@ -1,17 +1,33 @@
+import six
+
 from pycrunch.cubes import fetch_cube, count
 from scrunch.datasets import Variable
+from scrunch.expressions import parse_expr, process_expr
 from cr.cube.crunch_cube import CrunchCube
 
 
-def crtabs(dataset, variables):
+def crtabs(dataset, variables, weight=None, filter=None, **measures):
     """Return CrunchCube representation of crosstab.
 
     :param dataset: Dataset shoji object
     :param variables: List of variable urls, names or aliases
+    :param weight: Scrunch variable instance, alias or url
+    :param filter: Scrunch expression filter
     """
-    dataset = dataset.resource
     variables = prepare_variables(variables)
-    return CrunchCube(fetch_cube(dataset, variables, count=count()))
+    if weight:
+        if isinstance(weight, Variable):
+            weight = weight.url
+        elif isinstance(weight, six.string_types):
+            if weight in dataset.keys():
+                weight = dataset[weight].url
+        else:
+            raise KeyError('{} not found in dataset'.format(weight))
```

Comment:
@anderstheet  This line will raise a KeyError when the user passes a variable url, which should be supported as the docstring says.